templates/powershell/build/HelmLintAndPublish.ps1

```diff
@@ -28,9 +34,85 @@ param(
     [Parameter(Mandatory)]
     [string]$PSHelperDirectory,
     [Parameter(Mandatory)]
-    [string]$chartHomeDir
+    [string]$chartHomeDir,
+    [string]$KeyVaultVSecretNames,
+    [string]$ServiceName
 )
 
+
+function Update-KVSecretValues {
+    param(
+        [Parameter(Mandatory)]
+        [string]$InfraChartHomeDir,
+        [Parameter(Mandatory)]
+        [string]$ServiceName,
+        [Parameter(Mandatory)]
+        [string]$KeyVaultVSecretNames
+    )
+    begin {
+        [string]$functionName = $MyInvocation.MyCommand
+        Write-Debug "${functionName}:Entered"
+    }
+    process {
+        if (-not (Get-Module -ListAvailable -Name 'powershell-yaml')) {
+            Write-Host "powershell-yaml Module does not exists. Installing now.."
+            Install-Module powershell-yaml -Force
+            Write-Host "powershell-yaml Installed Successfully."
+        } 
+        else {
+            Write-Host "powershell-yaml Module exist"
+        }
+
+        $kvSecretNames = $KeyVaultVSecretNames | ConvertFrom-Json
+
+        Write-Debug "${functionName}:kvSecretNames:$kvSecretNames"
+    
+        $valuesYamlPath = "$InfraChartHomeDir\values.yaml"
+        [string]$content = Get-Content -Raw -Path $valuesYamlPath
+        Write-Debug "$valuesYamlPath content before: $content"
+        if ($content) {
+            $valuesObject = ConvertFrom-YAML $content -Ordered
+        }
+        else {
+            $valuesObject = [ordered]@{}
+        }
+
+        $keyVaultSecrets = [System.Collections.Generic.List[hashtable]]@()
+        foreach ($secret in $kvSecretNames) {
+                
+            #Logic to remove servicename from the secretname
+            #for e.g. "ffc-demo-payment-web-COOKIE-PASSWORD" will get replace with "COOKIE-PASSWORD"
+            if ($secret -like "$ServiceName*") {
+                $NoOfStartingCharsToTrunk = $ServiceName.Length + 1
+                $secretWithoutServiceName = $secret.subString($NoOfStartingCharsToTrunk, ($secret.Length - $NoOfStartingCharsToTrunk) )
+            }
+            else {
```

Comment:
I am unsure on this whole section, would want to discuss requirement in detail

Reply:
As discussed this is the existing logic, will be revisited during story 336179